tools/common/Target.cs

```diff
@@ -786,9 +786,15 @@ void GenerateIOSMain (StringWriter sw, Abi abi)
 					else
 						mono_native_lib = app.GetLibNativeName () + ".dylib";
 					sw.WriteLine ();
+#if NET
+					sw.WriteLine ($"\tmono_dllmap_insert (NULL, \"libSystem.Native\", NULL, \"{mono_native_lib}\", NULL);");
+					sw.WriteLine ($"\tmono_dllmap_insert (NULL, \"libSystem.Security.Cryptography.Native.Apple\", NULL, \"{mono_native_lib}\", NULL);");
+					sw.WriteLine ($"\tmono_dllmap_insert (NULL, \"libSystem.Net.Security.Native\", NULL, \"{mono_native_lib}\", NULL);");
```

Comment:
`libSystem.IO.Compression.Native.dylib` might be needed too.
It's shipped so it's likely used somewhere in the BCL (have not checked).

That can wait for the removal of the .dylib from the app bundle since it's the only way to test it properly.

Reply:
It's not needed for "Hello World" but it's needed in general. I hesitate to do more changes here since it's likely going to be replaced by `PINVOKE_OVERRIDE` callback anyway (unless there are some AOT implications from using dllmap).